src/passes/RemoveNonJSOps.cpp

```diff
@@ -97,6 +97,12 @@ struct RemoveNonJSOpsPass : public WalkerPass<PostWalker<RemoveNonJSOpsPass>> {
       }
       neededFunctions.clear();
     }
+
+    // Intrinsics may use memory, so ensure the module has one.
+    if (!module->memory.exists) {
+      module->memory.exists = true;
+      module->memory.initial = module->memory.max = 1;
+    }
```

Comment:
Maybe this should be a helper in module-utils.h?

Reply:
Good idea, adding.